iOSDFULibrary/Classes/Implementation/LegacyDFU/Characteristics/DFUControlPoint.swift

```diff
@@ -247,7 +251,11 @@ internal struct PacketReceiptNotification {
         self.resetSent = false
         
         // Get the peripheral object.
+        #if swift(>=5.5)
+        guard let peripheral = characteristic.service?.peripheral else { return }
```

Comment:
Should it just return without logging or execution of other logic in this method? Shouldn't it invoke/clear callbacks?

Reply:
True. I'll look into it today.

Reply:
I'll merge this PR and perhaps prepare another one with following changes.